### PR TITLE
Add auth:checkRights

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

Add auth:checkRights https://github.com/kuzzleio/sdk-jvm/issues/31

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### Other changes

Reverse wrong change in `build.gradle.kts`

### CODE REVIEW

**Review by scottinet — APPROVED** [review]

File: doc/1/controllers/auth/check-rights/index.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  ConcurrentHashMap<String, Object> requestPayload) throws NotConnectedException, InternalException
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `requestPayload` | <pre>ConcurrentHashMap<String, Object></pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
+
+## `requestPayload`
+
+The [RequestPayload](/core/2/api/payloads/request) must contains at least the following properties:
```
Comment (Leodau) [inline doc/1/controllers/auth/check-rights/index.md:29]:
```suggestion
The [RequestPayload](/core/2/api/payloads/request) must contain at least the following properties:
```

File: doc/1/controllers/auth/check-rights/index.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+- `controller`: API controller
+- `action`: API action
+
+---
+
+## Returns
+
+Returns a boolean telling whether the provided request would have been allowed or not
```
Comment (Leodau) [inline doc/1/controllers/auth/check-rights/index.md:38]:
```suggestion
Returns a boolean indicating whether the provided request would have been allowed or not
```

File: doc/1/controllers/auth/check-rights/index.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      requestPayload: ConcurrentHashMap<String, Any?>): CompletableFuture<Boolean>
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `requestPayload` | <pre>ConcurrentHashMap<String, Any?></pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
+
+## `requestPayload`
+
+The [RequestPayload](/core/2/api/payloads/request) must contains at least the following properties:
```
Comment (Leodau) [inline doc/1/controllers/auth/check-rights/index.md:58]:
```suggestion
The [RequestPayload](/core/2/api/payloads/request) must contain at least the following properties:
```

File: doc/1/controllers/auth/check-rights/index.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+title: checkRights
+description: Checks if the provided API request can be executed by this network connection, using the current authentication information.
+---
+
+# checkRights
+
+<SinceBadge version="Kuzzle 2.8.0"/>
+<SinceBadge version="auto-version"/>
+
+Checks if the provided API request can be executed by this network connection, using the current authentication information.
```
Comment (Leodau) [inline doc/1/controllers/auth/check-rights/index.md:13]:
```suggestion
Using the current authentication information, verify if the network connection can execute the required API request.
```
Heck why not..

File: doc/1/controllers/auth/check-rights/index.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
+---
+code: true
+type: page
+title: checkRights
+description: Checks if the provided API request can be executed by this network connection, using the current authentication information.
```
Comment (Leodau) [inline doc/1/controllers/auth/check-rights/index.md:5]:
```suggestion
description: Using the current authentication information, verify if the network connection can execute the required API request.
```

File: doc/1/controllers/auth/check-rights/index.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+The [RequestPayload](/core/2/api/payloads/request) must contains at least the following properties:
+
+- `controller`: API controller
+- `action`: API action
+
+---
+
+## Returns
+
+Returns a boolean telling whether the provided request would have been allowed or not
```
Comment (Leodau) [inline doc/1/controllers/auth/check-rights/index.md:67]:
```suggestion
Returns a boolean indicating whether the provided request would have been allowed or not
```

**Review by Leodau — APPROVED** [review]

**Review by jenow — APPROVED** [review]

**Pull request merged**